### PR TITLE
Fix Binary Compilation

### PR DESCRIPTION
**Pull request by scrogson**
This PR picks up from where #256 left off and adds some tests ~and tweaks the workspace configuration~.

~@evnu do the changes to the workspaces make sense?~

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -3,6 +3,4 @@ members = [
   "rustler",
   "rustler_codegen",
   "rustler_sys",
-  "rustler_tests/native/rustler_test",
```
Comment (evnu) [inline Cargo.toml]: I think I would not drop this. With this, running `cargo check` or similar in the root directory will also check the test crates, without having to run `mix test` in `rustler_tests` or `cargo check` for each of the test crates. For example, if something in `rustler_codegen` changes, problems are likely to be detected when running `cargo check` for `rustler_test`.

File: rustler_tests/native/deprecated_macros/Cargo.toml [hunk 2]
```diff
@@ -11,3 +11,5 @@ crate-type = ["cdylib"]
 
 [dependencies]
 rustler = { path = "../../../rustler" }
+
+[workspace]
```
Comment (evnu) [inline rustler_tests/native/deprecated_macros/Cargo.toml]: This is only a crate, but not a workspace. I think this is not needed.

File: rustler_tests/native/rustler_test/Cargo.toml [hunk 3]
```diff
@@ -12,3 +12,5 @@ crate-type = ["cdylib"]
 [dependencies]
 lazy_static = "1.4"
 rustler = { path = "../../../rustler" }
+
+[workspace]
```
Comment (evnu) [inline rustler_tests/native/rustler_test/Cargo.toml]: Same comment as [here](https://github.com/rusterlium/rustler/pull/261/files#r336749182), this is probably not needed?

**Comment by evnu** [comment]
@scrogson I think you forgot to add the new crate? :)

File: rustler_tests/native/deprecated_macros/Cargo.toml (hunk 2 above)
Comment (scrogson) [reply rustler_tests/native/deprecated_macros/Cargo.toml]: I added this to stop it from failing to think it was in a workspace. Based on your input above, I will remove this and add to the root workspace. :+1:

File: Cargo.toml (hunk 1 above)
Comment (scrogson) [reply Cargo.toml]: Makes sense :+1:

**Comment by scrogson** [comment]
> I think you forgot to add the new crate? :)

I messed up the `.gitignore`

**Comment by evnu** [comment]
@scrogson were you able to reproduce the error from #256 with the new crate? I still don't understand completely how the error is triggered. We use `[[bin]]` as well in our crates, but did not see any such error yet.

**Comment by scrogson** [comment]
> were you able to reproduce the error from #256 with the new crate?

Yes. If you remove the first commit and try running this with the changes I've made, you should see the error.



> We use `[[bin]]` as well in our crates, but did not see any such error yet.

It looks like if there is a `[lib]` section in the `Cargo.toml` then mix task skips the `[[bin]]` section entirely (might be nice to support both compiling libs and bins in the same crate). I noticed that when I added the `[[bin]]` section to the `rustler_test` crate, there was no failure. But, it also didn't move the binary into the `priv/native` dir.

**Comment by evnu** [comment]
This looks good then!

**Pull request merged**